### ISSUE TITLE
Quickfix window is no longer closed on Vim 7.4

### ISSUE THREAD

**Issue by cespare** [fatih/vim-go#1423]
After 9a433e5db072e767d10d7b66cf4807e1a10ba897 and the follow-up fix 52c5b1f74dade7ef21b34675751cd4a7b3ee2d03, the quickfix window is no longer closed in Vim 7.4.

To reproduce:

1. Open some go file.
2. Make it invalid by introducing a syntax error.
3. Save, so that a gofmt error appears in the quickfix window.
4. Fix the syntax error.
5. Save again, and observe that the quickfix window doesn't go away.

The particular vim I was using gave this for `vim --version`:

```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
Included patches: 1-1689
Extra patches: 8.0.0056
<lots of other junk>
```

The particular behaviors I observed were:

1. Before 9a433e5db072e767d10d7b66cf4807e1a10ba897: the quickfix window is closed when I expect.
2. After 9a433e5db072e767d10d7b66cf4807e1a10ba897: there's some crash regarding getqflist when the quickfix window is supposed to be closed.
3. After 52c5b1f74dade7ef21b34675751cd4a7b3ee2d03: there's no crash, but the quickfix window isn't closed.

I upgraded to vim 8.0.1033 and then this fixed the problem (now the quickfix window is closed as expected).

So basically, I think that 9a433e5db072e767d10d7b66cf4807e1a10ba897 needs to be adjusted for vim 7 compatibility.

**Comment by michaelyou** [comment]
I have also encountered this problem

**Comment by petesaia** [comment]
I can confirm as well. It appeared after an update.

**Comment by bhcleek** [comment]
@cespare @michaelyou @petesaia As a workaround you can also `:lclose` or `:cclose` (depending on your settings for the list to use for `:GoFmt`.

However, I think that #1474 will fix your issue. Can you confirm?

**Comment by jthurman42** [comment]
#1474 doesn't resolve this if you do not have patch 2200, but does resolve it with an additional line (commented in that PR)
